File: Week2 Assignment/MatrixTranspose.py
#Transpose of matrix


rows = int(input("Enter number of rows: "))
cols = int(input("Enter number of columns: "))

print("Enter matrix row-wise:")
matrix = [list(map(int, input().split())) for _ in range(rows)]


transpose = [[matrix[j][i] for j in range(rows)] for i in range(cols)]

print("Transpose of the matrix:")
for row in transpose:
    print(*row)

File: Week2 Assignment/MatrixTranspose.c
//Transpose of matrix

#include <stdio.h>

int main() {
    int rows, cols,i,j;

    printf("Enter number of rows and columns: ");
    scanf("%d %d", &rows, &cols);

    int matrix[rows][cols], transpose[cols][rows];

    
    printf("Enter elements of matrix:\n");
    for (i = 0; i < rows; i++)
        for (j = 0; j < cols; j++)
            scanf("%d", &matrix[i][j]);

    
    for (i = 0; i < rows; i++)
        for (j = 0; j < cols; j++)
            transpose[j][i] = matrix[i][j];

    
    printf("Transpose of the matrix:\n");
    for (i = 0; i < cols; i++) {
        for (j = 0; j < rows; j++)
            printf("%d ", transpose[i][j]);
        printf("\n");
    }

    return 0;
}

